### DISCUSSION TITLE
Organization-level custom domain not working for internal repository pages

### DISCUSSION THREAD

**Discussion by lpesola-fortum** [community/community#23350] [Other Feature Feedback, Questions, & Ideas]
<p>Hello there!</p>
<p>I’ve read the GitHub Pages custom domain docs and they state:</p>
<blockquote>
<p>After you configure a custom domain for a user or organization site, <strong>the custom domain will replace the <code>&lt;user&gt;.github.io</code> or <code>&lt;organization&gt;.github.io</code> portion of the URL for any project sites owned by the account that do not have a custom domain configured</strong>. For example, if the custom domain for your user site is <code>www.octocat.com</code> , and you have a project site with no custom domain configured that is published from a repository called <code>octo-project</code> , the GitHub Pages site for that repository will be available at <code>www.octocat.com/octo-project</code> .</p>
</blockquote>
<p>I have configured an organization-level custom domain by creating a repository with the name <code>organization</code> and by enabling pages and a custom domain <code>custom-subdomain.organization.com</code> for that.</p>
<p>Based on the documentation above I was expecting that from now on all pages that do not specify another custom domain would be found under <code>custom-subdomain.organization.com/repository-name</code> but that is not the case. Instead, I get 404 " The site configured at this address does not contain the requested file. "<br>
What am I missing? Is there any way to enable a custom (sub)domain for an entire organization such that all pages would automatically use that domain as explained in the docs?</p>
<p>Repository-specific custom domains work as intended, but setting up a new one for each repository is not a feasible solution. Redirection from <code>organization.github.io/repo-name</code> also works but that too is not an acceptable solution since it redirects to the default (unusable) domain name that all private pages have (unless a repository-specific custom domain is set).</p>

**Comment by yoannchaudet** [comment]
<p>Hello and welcome here!</p>
<p>The behavior you are describing works as intended but only for public Pages sites.</p>
<p>For security reasons, private Pages sites (the ones with a random default hostname that enforce authentication + authorization on access) need to be isolated.</p>

**Comment by lpesola-fortum** [comment]
<p>Thanks for the clarification! If I may suggest, think it would be useful to have a specific section in the docs where the limitations of private/internal pages are explained :slight_smile:</p>
